apm-agent-plugins/apm-servlet-plugin/src/main/java/co/elastic/apm/agent/servlet/ServletTransactionHelper.java

```diff
@@ -169,6 +170,7 @@ private void doOnAfter(Transaction transaction, @Nullable Throwable exception, b
         }
         fillResponse(transaction.getContext().getResponse(), committed, status);
         transaction.withResultIfUnset(ResultUtil.getResultByHttpStatus(status));
+        transaction.withOutcome(status < 500 ? Outcome.SUCCESS : Outcome.FAILURE);
```

Comment:
Is it feasible to set the outcome in a `co.elastic.apm.agent.report.processor.Processor` or on `co.elastic.apm.agent.impl.transaction.AbstractSpan#end()` to consistently set the outcome irrespective of the web/client/... framework instrumentation?

Reply:
Yes, that could definitely be a good shortcut to handling all the HTTP plugins at once.

I'm not sure about the processor approach, as it would change state of spans & transactions between the time they are reported and when they are sent, which might complicate a bit testing.